web/src/components/bgImagewithCard.js

```diff
@@ -7,25 +7,29 @@ import { convertToBgImage } from 'gbimage-bridge';
 
 import Card from './card';
 
-export default function BgImagewithCard({
-    data,
-    hasButton,
-    isDark,
-    isRight,
-    isTransparent,
-    isPrimaryDark,
-    isPrimary,
-    isSecondaryDark,
-    isSecondary,
-    to,
-    href,
-}) {
+export default function BgImagewithCard({ data }) {
     const {
         backgroundImage: { asset },
+        cardType,
+        backgroundColor,
+        cardOrientation,
+        button,
+        link,
+        linkType,
     } = data;
     const imageData = getImage(asset);
     const bgImageData = convertToBgImage(imageData);
 
+    const isDark = backgroundColor === 'dark';
+    const isTransparent = backgroundColor === 'transparent';
+    const isRight = cardOrientation === 'right';
+    const hasButton = button === 'button';
+    const isPrimary = cardType === 'primary';
+    const isPrimaryDark = cardType === 'secondary';
+    const isSecondaryDark = cardType === 'transparent';
+    const externalLink = linkType === 'external';
+    const internalLink = linkType === 'internal';
```

Comment:
I'd like to see these cleaned up. I don't think we need `isPrimaryDark` and `isSecondaryDark` options. `hasButton` can be evaluated from the `link` field.

Reply:
yes we do, those are the colors we need, per our convo

Reply:
`isPrimary/isSecondary` and `isLight/isDark` aren't enough?

Reply:
is primary and issecondary have to do with button colors. we need to add for transparent (if user wants transparent button). technically its called secondaryDark but if you want me to rename it to transparent button happy to do so but yes I think its necessary.

Reply:
I'm not sure I'm following. I think we should be able to derive all of the stying we need with less props. I think all we need are `isPrimary`, `isSecondary`, `isTransparent`, `isDark`. Adding props like `isPrimaryDark` is superfluous.

Reply:
What prop determines the background color of the card?

Reply:
It's not superfluous it is the button for the transparent card.

Reply:
isDark determines the color for the card.

Reply:
Do we need `isPrimaryDark` too though? That's the question I'm asking.

Reply:
yeah, we are assigning that color scheme - primary dark from the mock ups as the secondary button color --dark card with light button

Reply:
I guess what I'm asking is, can we apply styling if we only have `isPrimary` and `isDark` props?. Do we have to have unique `isPrimaryDark` etc? 

IMHO, the props/css as it is now is very confusing to me. This seems like a good opportunity to refactor to something simpler that will hopefully help future devs.

Reply:
We have a card that can be dark light or transparent. This is set with isDark or isTransparent. Then the card has it's adjoining button, primary (set with the primarybutton) secondary (set with isprimarydark) or transparent (set with secondaryDark). We spoke about this yesterday needing the third button for the transparent card. I can flesh out the description and write up some documentation to make this more clear?